core/modules/wbase/Task.cc

```diff
@@ -150,18 +150,29 @@ std::vector<Task::Ptr> Task::createTasks(std::shared_ptr<proto::TaskMsg> const&
                 }
             } 
             auto task = std::make_shared<wbase::Task>(taskMsg, sendChannel);
-            task->setQueryStr(qs); //&&& remove and make part of constructor
+            task->setQueryString(qs); //&&& remove and make part of constructor
             task->setQueryFragmentNum(fragNum); //&&& Is it better to move fragment info from
                                           //&&& ChunkResource getResourceFragment(int i) to here???
                                           //&&& It looks like Task should contain a ChunkResource::Info object
                                           //&&& which could help cleaan up the mess in ChunkResource.
             vect.push_back(task);
         }
     }
+    sendChannel->setTaskCount(fragmentCount);
     return vect;
 }
 
 
+void Task::setQueryString(std::string const& qs) {
+    _queryString = qs;
```

Comment:
Is this safe to make this modification w/o doing any locking on the object state?

Reply:
From an earlier commit before this was moved into the constructor.